installer/scripts/tue-install-impl.bash

```diff
@@ -291,17 +291,37 @@ function tue-install-cp
             root_required=false
         fi
 
-        if ! cmp --quiet "$file" "$2"
+        if [ -d "$2" ]
         then
-            tue-install-debug "File $file and $2 are different, copying..."
-            if "$root_required"
+            cp_target="$2/"$(basename "$file")
+
+            if ! cmp --quiet "$file" "$cp_target"
             then
-                tue-install-debug "Using elevated privileges (sudo)"
-                sudo mkdir --parents --verbose "$(dirname "$2")" && sudo cp --verbose "$file" "$2"
-            else
-                mkdir --parents --verbose "$(dirname "$2")" && cp --verbose "$file" "$2"
+                tue-install-debug "File $file and $cp_target are different, copying..."
+                if "$root_required"
+                then
+                    tue-install-debug "Using elevated privileges (sudo)"
+                    sudo mkdir --parents --verbose "$2" && sudo cp --verbose "$file" "$cp_target"
+                else
+                    mkdir --parents --verbose "$2" && cp --verbose "$file" "$cp_target"
+                fi
+            fi
+        else
+            cp_target="$2"
```

Comment:
Define all the variable in a IF/ELSE, then do de comparison and execution. This reduces double code. I haven't tested the code at the moment.

Reply:
Have a look. The only other thing I can think of is the creation of the target directory outside the loop as it will happen only once. Should I do that or leave it as it is?

Reply:
I think it looks good.